docs/source/models.rst

```diff
@@ -307,49 +236,136 @@ model artifacts, together with other model metadata. To set a signature on a log
-Params schema is a sequence of ParamSpec, each of which contains ``name``, ``type``, ``default`` and ``shape`` fields.
-``type`` field must be specified as one of the :py:class:`MLflow data types <mlflow.types.DataType>`, and ``shape`` 
-field should be ``None`` for scalar parameters, or ``(-1,)`` for list parameters.
-See some examples of constructing them below.
+A model signature consists of three schemas **(1) inputs**, **(2)outputs**, and **(3) params**. The
+former two refer to schemas of model inputs and outputs, while `params` defines schema for additional params 
+that are passed to the model at inference time. Input and output schemas can be either column-based or tensor-based. 
 
-Column-based Signature Example
-""""""""""""""""""""""""""""""
-Each column-based input and output is represented by a type corresponding to one of
-:py:class:`MLflow data types <mlflow.types.DataType>` and an optional name. Input columns can also be marked
-as ``optional``, indicating whether they are required as input to the model or can be omitted. The following example
-displays a modified MLmodel file excerpt containing the model signature for a classification model trained on
-the `Iris dataset <https://archive.ics.uci.edu/ml/datasets/iris>`_. The input has 4 named, numeric columns and 1 named,
-optional string column.
-The output is an unnamed integer specifying the predicted class.
+Column-based Signature
+~~~~~~~~~~~~~~~~~~~~~~
+This signature is typically used for traditional machine learning models that take tabular data as input such as Pandas 
+DataFrame. Column-based schemas are a sequence of (optionally) named columns with data type. Each column-based input and 
+output is represented by a type corresponding to one of :ref:`supported data types <supported-data-types-column>` and an 
+optional name. Input columns can also be marked as ``optional``, indicating whether they are required as input to the 
+model or can be omitted. 
+
+The following example displays the model signature for a classification model trained on the 
+`Iris dataset <https://archive.ics.uci.edu/ml/datasets/iris>`_. The input has 4 named, numeric columns and 
+1 named, optional string column. The output is an unnamed integer specifying the predicted class.
 
 .. code-block:: yaml
 
   signature:
-      inputs: '[{"name": "sepal length (cm)", "type": "double"}, {"name": "sepal width
-        (cm)", "type": "double"}, {"name": "petal length (cm)", "type": "double"}, {"name":
-        "petal width (cm)", "type": "double"}, {"name": "class", "type": "string", "optional": "true"}]'
+      inputs: '[
+        {"name": "sepal length (cm)", "type": "double"}, 
+        {"name": "sepal width(cm)",   "type": "double"}, 
+        {"name": "petal length (cm)", "type": "double"}, 
+        {"name": "petal width (cm)",  "type": "double"}, 
+        {"name": "class",             "type": "string",  "optional": "true"}
```

Comment:
```suggestion
        {"name": "class",             "type": "string",  "required": "true"}
```

do we need to change optional to required?

Reply:
Oh we might need to add 'required': true to every line

Reply:
> do we need to change optional to required?

Ah yes, will do.

Btw I also think we should remove that field. It's the target class we want to predict:p https://archive.ics.uci.edu/dataset/53/iris